backend/services/api_utility_service.py

```diff
@@ -0,0 +1,96 @@
+import json
+from typing import Generic, T
+
+from fastapi.encoders import jsonable_encoder
+from pydantic import parse_obj_as, TypeAdapter
+from pydantic.v1.main import ModelMetaclass
+from sqlalchemy import TypeDecorator, JSON
+from sqlmodel import Session, create_engine
+
+from services.config_service import config
+
+
+dbUrl = f'postgresql://postgres:{config.postgres_password}@db:5432/postgres'
+engine = create_engine(dbUrl)
+
+
+def get_session():
+    with Session(engine) as session:
+        yield session
+
+
+# This method borrowed from Maximilian Franz
+# Code located here: https://github.com/tiangolo/sqlmodel/issues/63
+def pydantic_column_type(pydantic_type):
```

Comment:
I don't see where this is being used but maybe I missed that in my review.

Reply:
This is used in the Course model definition in the backend when creating the Node attribute:
```
nodes: Optional[List[Node]] = Field(default=None, sa_column=Column(pydantic_column_type(Optional[List[Node]])))
```

It will need to be used anytime we want to use a custom/pydantic-based type for database attributes. It ensures that the content is serialized correctly when reading data from the database without additional code, like in the test cases:
```
for node in test_course.nodes:
    node_names.append(node.title)
```

Writing data to fields with custom type still require a ``` model_dump() ``` method. Haven't found a way around this yet. Example in another test case:
```
test_nodes.append(Node(title=f'Node {i}', short_description=f'Node {i}').model_dump())
```

Reply:
Oh I see now. In that case it looks good. Hopefully we can find the model_dump() work around but if not that's fine too.